### PR TITLE
Fix multiple issues to allow "mgr-inter-sync" to run successfully

### PR DESCRIPTION
**Pull request by meaksh**
## What does this PR change?

This PR fixes several issues found in order to make `mgr-inter-sync` to run succesfully on Python 3:

- Fix wrong references to unexisting `suseRepositories`, `suseProductRepository` and `suseProductSCCRepositories` tables.
- Fix wrong attributes on XML schema construction during satellite exporter.
- On long SSL frames, read SSL decoded buffer completely even if no pending bytes on the underlying connection (avoid crashing due malformed XML).
- Fix issues with the `CompressedCache` due buffer encoding.
- Relative imports.

Now, with this PR, `mgr-inter-sync` seems to run successfully (including the package downloading):
```console
suma40-head-iss-slave:~ # mgr-inter-sync -c testchannel --email                                                                    
10:42:17 SUSE Manager - live synchronization                                                                                                                                
10:42:17    url: https://suma40-head-iss-master.tf.local                          
10:42:17    debug/output level: 8                                                     
10:42:17    +++ SUSE Manager Server synchronization tool checking in.                      
10:42:18    +++ Entitled SUSE Manager Server validated.                                   
10:42:18    db:  spacewalk/<password>@susemanager                                  
10:42:18 Action list/commandline toggles: ['orgs', 'channel-families', 'arches', 'channels', 'short', 'cloned-channels', 'download-packages', 'rpms', 'packages', 'download-
errata', 'download-kickstarts', 'errata', 'kickstarts', 'supportinfo', 'suse-products', 'scc-repositories', 'suse-product-channels', 'suse-upgrade-paths', 'suse-product-ext
ensions', 'suse-product-repositories', 'suse-subscriptions']                                    
10:42:18                                                                                       
10:42:18 Retrieving / parsing orgs data                                            
10:42:18 orgs data complete                                                       
10:42:18                                                                         
10:42:18 Retrieving / parsing channel-families data                                
10:42:19    parsing family: SUSE (1) Channel Family                               
10:42:19    parsing family: SUSE Linux Enterprise High Availability Extension (PPC)   
10:42:19    parsing family: SUSE Linux Enterprise High Availability Extension (PPC) (ALPHA)
10:42:19    parsing family: SUSE Linux Enterprise High Availability Extension (PPC) (BETA)
10:42:19    parsing family: SUSE Linux Enterprise High Performance Computing (x86)
10:42:19    parsing family: SUSE Linux Enterprise High Performance Computing (x86) (ALPHA)
10:42:19    parsing family: SUSE Linux Enterprise High Performance Computing (x86) (BETA)   
10:42:19    parsing family: SUSE Linux Enterprise Point of Service                                              
...
10:42:19 channel-families data complete                                                   
10:42:19                                                        
10:42:19 Retrieving / parsing product names data                        
10:42:19 product names data complete                                   
10:42:19                                                          
10:42:19 Retrieving / parsing arches data                                 
10:42:19    parsed arch: aarch64                                         
10:42:19    parsed arch: alpha                                
10:42:19    parsed arch: alphaev6                                     
10:42:19    parsed arch: amd64                                       
...
10:42:19 arches data complete
10:42:19
10:42:19 Retrieving / parsing additional arches data
10:42:20 additional arches data complete
10:42:20
10:42:20 Retrieving / parsing channel data
XXX: imported channels: ['sles12-sp3-pool-x86_64']
XXX:   cached channels: ['testchannel', 'sles12-sp3-pool-x86_64', 'sle-manager-tools12-pool-x86_64-sp3', 'sle-manager-tools12-updates-x86_64-sp3', 'sles12-sp3-updates-x86_6
4']
10:42:20    p = previously imported/synced channel
10:42:20    . = channel not yet imported/synced
10:42:20    base-channels:
10:42:20       . testchannel                                14       full import from Wed Feb 27 10:38:28 2019
10:42:20
10:42:20
10:42:20 Syncing Channel testchannel to Org 1
10:42:20 Channel data complete
10:42:20
10:42:20 Retrieving short package metadata (used for indexing)
10:42:20    Retrieving / parsing short package metadata: testchannel (14)
10:42:20 Diffing package metadata (what's missing locally?): testchannel
            ________________________________________
Diffing:    ######################################## - complete
10:42:20
10:42:20 Retrieving / parsing cloned_channels data
10:42:20 cloned_channels data complete
10:42:20
10:42:20 Downloading package metadata
10:42:20    Retrieving / parsing *relevant* package metadata: testchannel (NONE RELEVANT)
10:42:20
10:42:20 Downloading rpm packages
10:42:20    Fetching any missing RPMs: testchannel (NONE MISSING)
10:42:20 Processing rpm packages complete
10:42:20
10:42:20 Importing package metadata
10:42:20    Importing *relevant* package metadata: testchannel (NONE RELEVANT)
10:42:20
10:42:20 Linking packages to channels
10:42:20
10:42:20 Downloading patch data
10:42:20    Retrieving / parsing errata data: testchannel (NONE RELEVANT)
10:42:20 Downloading patch data complete
10:42:20
10:42:20 Downloading kickstartable trees metadata
10:42:20    Retrieving / parsing kickstart data: testchannel (NONE RELEVANT)
10:42:20
10:42:20 Downloading kickstartable trees files
10:42:20    Retrieving / parsing kickstart tree files: testchannel (NONE RELEVANT)
10:42:20
10:42:20 Importing channel patches
10:42:20 Importing 0 patches for channel testchannel.
10:42:20    Importing *relevant* errata: testchannel (NONE RELEVANT)
10:42:20    No new kickstartable tree to import
10:42:20
10:42:20 Retrieving / parsing supportinfo data
10:42:31 supportinfo data complete
10:42:31
10:42:31 Retrieving / parsing suse_products data
10:42:40 suse_products data complete
10:42:40
10:42:40 Retrieving / parsing scc_repositories data
10:42:45 scc_repositories data complete
10:42:45
10:42:45 Retrieving / parsing suse_product_channels data
10:42:45 suse_product_channels data complete
10:42:45
10:42:45 Retrieving / parsing suse_upgrade_paths data
10:42:46 suse_upgrade_paths data complete
10:42:46
10:42:46 Retrieving / parsing suse_product_extensions data
10:42:48 suse_product_extensions data complete
10:42:48
10:42:48 Retrieving / parsing suse_product_repositories data
10:43:05 suse_product_repositories data complete
10:43:05
10:43:05 Retrieving / parsing suse_subscriptions data
10:43:05 suse_subscriptions data complete
    Import complete:
        Begin time: Wed Feb 27 10:42:17 2019
        End time:   Wed Feb 27 10:43:05 2019
        Elapsed:    0 hours, 0 minutes, 47 seconds

+++ sending log as an email +++
suma40-head-iss-slave:~ #
```

## GUI diff

No difference.

- [x] **DONE**

## Documentation
- No documentation needed: **python3 porting**

- [x] **DONE**

## Test coverage
- No tests: **python3 porting**

- [x] **DONE**

## Links

Tracks https://github.com/SUSE/salt-board/issues/247

- [x] **DONE**

### CODE REVIEW

**Comment by mcalmer** [comment]
@meaksh forget my last comment. I cannot remember my own code :-) I need some time to remember everything again

File: backend/server/importlib/backend.py [hunk 1]
```diff
@@ -1499,39 +1499,39 @@ def processSuseProductRepositories(self, batch):
                AND repo_id = :repo_id
         """)
         _query_pr = self.dbmodule.prepare("""
-            SELECT product_id, root_product_id, repo_id FROM suseProductRepository
+            SELECT product_id, root_product_id, repo_id FROM suseProductSCCRepository
             """)
         _query_pr.execute()
-        existing_data = ["%s-%s-%s" % (x['product_id'], x['root_product_id'], x['repo_id']) for x in _query_pr.fetchall_dict() or []]
+        existing_data = ["%s-%s-%s" % (x['product_pdid'], x['root_pdid'], x['repo_pdid']) for x in _query_pr.fetchall_dict() or []]
```
Comment (mcalmer) [inline backend/server/importlib/backend.py]: I think the original was correct. `_query_pr` is 

```
SELECT product_id, root_product_id, repo_id FROM suseProductSCCRepository
```

So the field names are product_id, root_product_id, repo_id.

File: backend/server/importlib/backend.py [hunk 2]
```diff
@@ -1499,39 +1499,39 @@ def processSuseProductRepositories(self, batch):
...
         toinsert = [[], [], [], [], [], [], [], [], []]
         todelete = [[], [], []]
         toupdate = [[], [], [], [], [], [], [], []]
         for item in batch:
-            ident = "%s-%s-%s" % (item['product_id'], item['root_id'], item['repo_id'])
+            ident = "%s-%s-%s" % (item['product_pdid'], item['root_pdid'], item['repo_pdid'])
```
Comment (mcalmer) [inline backend/server/importlib/backend.py]: Not sure here. It depends what are the keys in "batch"

Comment (meaksh) [reply backend/server/importlib/backend.py]: This is correct in this particular case.

**Pull request merged**